Payloads/Payload.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace Concept.Payloads
+{
+    /// <summary>
+    /// Represnts a generic payload.
+    /// </summary>
+    public class Payload
+    {
+        [JsonProperty("op")]
+        public string Op { get; set; }
```

Comment:
We need to have Enums for that so the user know what OP types are expected.

Reply:
Wouldn't users be interacting purely with WS and mainly send JSON strings? Perhaps the Enum for OP types could be in the base client library.

Reply:
Yes they would be but they dont know what op codes are handled by us. So having enums would make it simpler and their int codes for each enum value. Some people just want fast wrappers and might do stuff with bytes.

Reply:
That's fair

Reply:
So here are some changes I made: 

```csharp
public IPayload GetPayloadFromBuffer(ReadOnlyMemory<byte> buffer)
{
            var reader = new Utf8JsonReader(buffer.Span);

            IPayload payload;

            Console.WriteLine(reader.TokenType.ToString());

            if (reader.TokenType != JsonTokenType.StartObject)
                throw new JsonException("Websocket did not receive a JSON payload.");

            for(int i = 0; reader.TokenType != JsonTokenType.EndObject; i++)
            {
                PayloadOp op;

                //Console.WriteLine(i);
                //Console.WriteLine(reader.TokenType.ToString());

                if(i == 2)
                {
                    //Check if op was either sent as a number or a string
                    if(reader.TokenType == JsonTokenType.Number)
                        switch(reader.GetInt32())
                        {
                            case 1:
                                op = PayloadOp.Connect;

                                payload = new ConnectPayload();

                                BuildConnectPayload(ref payload, ref reader);

                                payload.Op = op;

                                break;

                            case 2:
                                //Repeat the above case for every case
                                op = PayloadOp.Play;
                                break;

                            case 3:
                                op = PayloadOp.Stop;
                                break;

                                //Continue this for all payloads
                        }
                    else if(reader.TokenType == JsonTokenType.String)
                        switch(reader.GetString())
                        {
                            case "connect":
                                op = PayloadOp.Connect;
                                break;

                                //Continue this for all payloads
                        }

                    break;
                }

            }

        }

        public void BuildConnectPayload(ref IPayload payload, ref Utf8JsonReader reader)
        {
            var connPayload = new ConnectPayload();
            //Use builder for further byte parsing here.
}
```

The builder methods would be repeated for each builder. I also added string and number support as separate switch statements incase a user sends that in a payload. I'm also doubting that we need an Op property in these payload classes. Let me know your thoughts on this.

Also, I assume parsing through bytes is generally faster than just parsing through a string of data?

Reply:
Wouldn't that fail?

Reply:
The number op?

Reply:
Matching ops and building objects